src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -146,8 +146,13 @@ public void updateUserPassword(String username, String password) {
     }
 
     @Override
-    public void updateUserEmail(String username, String email) {
-        userRepository.saveUserEmailByUsername(username, email);
+    public void updateUserEmail(String username, String email, UUID codeId) {
+        userRepository.findByUsername(username).ifPresent((u)->{
```

Comment:
You should create a new SQL request with updating selected fields instead of fetching whole the user.